src/app/main-page/presentation/toolbar-action/toolbar-action.component.ts

```diff
@@ -56,7 +60,7 @@ export class ToolbarActionComponent implements OnChanges {
     exampleData: ExampleData[] = Constants.EXAMPLE_DATA_FILE_STRUCTURE;
 
     constructor(
-        private mainPageService: MainPageService,
+        private ioService: IOService,
```

Comment:
This is a view component. Services should be used in container components.

Reply:
Everything is moved to the container component, presentation component is really dumb now.